### PR TITLE
Run3-alca204 Modernize AlCaDiJetsProducer.cc

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Modernize AlCaDiJetsProducer.cc

#### PR validation:

Use the runTheMatrix test workflows 

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Nothing special

### CODE REVIEW

**Comment by tvami** [comment]
@cmsbuild , please test

**Comment by tvami** [comment]
@cmsbuild , please test

**Comment by tvami** [comment]
@cmsbuild please test

**Comment by tvami** [comment]
+alca

File: Calibration/HcalAlCaRecoProducers/plugins/AlCaDiJetsProducer.cc [hunk 1]
```diff
@@ -32,26 +33,36 @@
...
+    mutable std::atomic<unsigned int> nAll_, nSelect_;
+  };
+}  // namespace alcaHcalDiJet
+
 //
 // class declaration
 //
 
-class AlCaDiJetsProducer : public edm::EDProducer {
+class AlCaDiJetsProducer : public edm::global::EDProducer<edm::RunCache<alcaHcalDiJet::Counters>> {
```
Comment (tvami) [inline Calibration/HcalAlCaRecoProducers/plugins/AlCaDiJetsProducer.cc:47]: why is this prefered to be a `global` instead of a `one` module?

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by tvami** [comment]
Hi @bsunanda I think you picked up Marco's commit about the seed generators, that shouldnt be part of this PR, right?

File: Calibration/HcalAlCaRecoProducers/plugins/AlCaDiJetsProducer.cc (hunk 1 above)
Comment (mmusich) [reply Calibration/HcalAlCaRecoProducers/plugins/AlCaDiJetsProducer.cc:47]: According to the [manual](https://twiki.cern.ch/twiki/bin/view/CMSPublic/FWMultithreadedFrameworkModuleTypes#One_Modules)

> A One module is a module which, for what ever reason, is unable to be used by multiple threads simultaneously and is therefore thread-unsafe. Like a Global module, one instance of a One module is shared across all Streams. However, unlike a Global module, the framework enforces thread-safety by only allowing one Event, LuminosityBlock or Run at a time to be passed to a One module instance. The use of One modules is terrible for performances.

Comment (tvami) [reply Calibration/HcalAlCaRecoProducers/plugins/AlCaDiJetsProducer.cc:47]: Thanks @mmusich I thought that the ALCARECOs are to be produces single-threaded, i.e. as a `one` module, so then this DiJetsProducer is an exception?

Comment (mmusich) [reply Calibration/HcalAlCaRecoProducers/plugins/AlCaDiJetsProducer.cc:47]: I think only express is configured to be single-threaded. In Prompt everything is multi-threaded.

**Comment by bsunanda** [comment]
@tvami I wonder how it came about. Most likely I had one extra thing. It is messed up - so closing this PR

**Comment by bsunanda** [comment]
Closing it in favor of #35448

**Comment by tvami** [comment]
-alca
 * PR is to be closed

**Comment by bsunanda** [comment]
Close this PR in favor of #35448

**Pull request closed without merging**